File: python/28_count_consonants.py
"""Contar consonantes — Fácil · Strings

Cuenta las consonantes a-z/A-Z e ignora números, espacios y puntuación.

Fuente: https://coding-challenges.dev/problems/python-contar-consonantes
"""


def count_consonants(text: str) -> int:
    vowels = set("aeiouAEIOU")
    return sum(char.isascii() and char.isalpha() and char not in vowels for char in text)

File: typescript/28-count-consonants.ts
// Contar consonantes — Fácil · Strings
// Cuenta consonantes a-z/A-Z e ignora caracteres no alfabéticos.
// Fuente: https://coding-challenges.dev/problems/contar-consonantes

function countConsonants(text: string): number {
  return (text.match(/[b-df-hj-np-tv-z]/gi) ?? []).length;
}

// No modificar: necesario para evaluar el resultado.
export { countConsonants };
